submitter/relayer/relayer.go

```diff
@@ -293,12 +308,85 @@ func (rl *Relayer) calculateBumpedFee(ckptInfo *types.CheckpointInfo) btcutil.Am
 	// Calculate the recommended fee using ResubmitFeeMultiplier
 	bumpedFee := ckptInfo.Tx2.Fee.MulF64(rl.config.ResubmitFeeMultiplier)
 
+	if previousFailure == nil ||
+		(!errors.Is(previousFailure, ErrInsufficientFee) &&
```

Comment:
maybe we can move this three error checks to separate function with the name ? (it is a bit hard to read this thing 😅 )